src/index.js

```diff
@@ -109,119 +110,56 @@ async function retrievePDF(order) {
 // even if this in another domain here, but it works as an example
 async function authenticate(username, password) {
   const xsrfToken = await getXsrfToken()
-  const loginResponse = await request({
+  const options = {
+    xsrfToken,
+    username,
+    password,
+    url: `${baseURL.origin}/api/reef/login`,
+    validate: ({ body: { status } }) => status
+  }
+
+  const isSigned = await signin(options)
+  if (!isSigned) {
+    throw new Error('Unable to signin, check the username and/or the password')
+  }
+}
+
+async function signin({ xsrfToken, username, password, url, validate }) {
+  const req = require('request-promise')
+  const response = await req({
     jar,
     headers: {
       'x-xsrf-token': xsrfToken,
-      'accept': 'application/json',
+      accept: 'application/json',
       'content-type': 'application/json'
     },
     body: {
       username,
       password,
-      'wnaMode': 'shop'
+      wnaMode: 'shop'
     },
     json: true,
     method: 'POST',
-    url: 'https://www.zalando.fr/api/reef/login',
-    resolveWithFullResponse: false
-  })
-
-  const data = await request({
-    jar,
-    method: 'GET',
-    url: 'https://www.zalando.fr/moncompte/commandes/1/'
+    uri: url,
+    resolveWithFullResponse: true
   })
+  return validate(response)
+}
 
-  const $accountPage = cheerio.load(data)
+// convert a price string to a float
+function normalizePrice(text) {
+  return parseFloat(/\d+,\d+/.exec(text)[0].replace(',', '.'))
+}
 
-  const orders = scrape($accountPage, {
-    commandNum: {
-      sel: 'td.order',
-      parse: (content) => content.substr(2)
-    },
-    date: {
-      sel: 'td.date'
-    },
-    total: {
-      sel: 'td.oTotal',
-      parse: normalizePrice
-    },
-    isPaid: {
-      sel: 'td.statusPayment',
-      parse: (content) => content.indexOf('Réglé') !== -1 ? true : false
-    },
-    isShipped: {
-      sel: 'td.statusShipping',
-      parse: (content) => content.indexOf('Envoyée') !== -1 ? true : false
-    },
-    orderId: {
-      sel: 'td.nextStep a',
-      attr: 'href',
-      parse: (content) => content.match(/\d+/g)
-    }
-
-  }, '#myOrdersTable tbody tr.cWrapper')
-
-  const pdfBills = Promise.all(orders.map(retrievePDF))
-  return await saveBills(pdfBills, 'test', {
-    identifiers: ['zalando'],
-    contentType: 'application/pdf'
-  })
+function createFilename({ numero, amount }) {
+  return `${numero}-${amount}-EUR.pdf`
```

Comment:
starting with YYYYMMDD is cool for sorting

Reply:
To homogenize all this, the choice should not be left to the third party developer.

Reply:
There should at least be a default filename value in saveBills, calculated with the bill's mandatory attributes.